### ISSUE TITLE
Map.merge and warning despite @Nullable

### ISSUE THREAD

**Issue by rionda** [typetools/checker-framework#7082]
I have code that looks like this:

```java

class MyClass {

    void method() { 

        Map<MyKey,Integer> a = new HashMap<>(); // I'm not using a HashMap, but something that implements Map
        ...
        a.merge(key, -1, ThisClass::mergeFunction);
        ...
    }

    private static @Nullable Integer mergeFunction(Integer old, Integer v) {
        if (old + v == 0) {
            return null;
        } else {
            return old + v;
        }
    }
}
```
When I try to compile, I get the following message:
```
java: [methodref.return] Incompatible return type
  found   : @Initialized @Nullable Integer
  required: @Initialized @NonNull Integer
  Consequence: method in @Initialized @NonNull MyClass
    @Initialized @Nullable Integer mergeFunction(@Initialized @NonNull Integer p0, @Initialized @NonNull Integer p1)
  is not a valid method reference for method in @Initialized @NonNull BiFunction<@Initialized @NonNull Integer, @Initialized @NonNull Integer, @Initialized @NonNull Integer>
    @Initialized @NonNull Integer apply(@Initialized @NonNull BiFunction<@Initialized @NonNull Integer, @Initialized @NonNull Integer, @Initialized @NonNull Integer> this, @Initialized @NonNull Integer p0, @Initialized @NonNull Integer p1)
```

Is this a false positive, and if not, can any annotation be used to guide the nullness checker? Or shall I just suppress the warning?

**Comment by rionda** [comment]
Thanks for addressing this. I think the same issue exists also for the various `compute*` methods of `Map`.

**Comment by mernst** [comment]
You are right.  (Not all `compute*` methods, but all the ones that take a remapping function as an argument.)  Thanks!  I have fixed them too, in https://github.com/typetools/jdk/pull/227.
